### ISSUE TITLE
Setting the x axis labels via pandas dataframe won't carry to mpld3

### ISSUE THREAD

**Issue by joaodlf** [mpld3/mpld3#364]
Hi, 

I've made an example on this notebook:
https://tmp49.tmpnb.org/user/wvAjc7U0cb38/notebooks/notebook.ipynb

Basically, my x axis labels don't seem to carry over when I do mpld3.display(fig). The labels display correctly before mpld3 kicks in (Google, Facebook, etc...)

The same thing happens to the legend of the plot. What is the correct way to fix these issues?

**Comment by jakevdp** [comment]
mpld3 does not currently support fixed formatters for axes, which pandas uses extensively.

**Comment by joaodlf** [comment]
Is there any possibility at all to display the labels correctly without having to hack away into javascript?

**Comment by joaodlf** [comment]
Or even, an alternative to mpld3 that would be able to accommodate this?

**Comment by jakevdp** [comment]
> Is there any possibility at all to display the labels correctly without having to hack away into javascript?

There may be some improvement on mpld3 master.

> even, an alternative to mpld3 that would be able to accommodate this?

Not that I'm aware of. Bokeh is a good solution, but its matplotlib importer uses the same infrastructure as mpld3, so I don't think it will give much of an improvement.

**Comment by bryevdv** [comment]
Yes, if using current MPL compat, Bokeh will have the same issues. But you can probably accomplish what you want if used native Bokeh APIs.

 Some day we will all get MEP 25 :)

**Comment by joaodlf** [comment]
Bokeh does accomplish this through a simple
`p = Bar(df_grouped, label='tracking_medium', values='revenue_standard')`

It's a shame really, overall this does looks like a much better implementation - I'd much rather use mpld3 if it weren't for this one hiccup!

**Comment by bryevdv** [comment]
Sorry it works for you, I guess? :) Seriously though, hopefully MEP 25 will be a priority after MPL 2.0 and then all the various python viz projects can benefit from consistent, stable, and comprehensive MPL compatibility.

**Comment by joaodlf** [comment]
I'm sorry if my comment came across as rude! It's just my opinion really, mpld3 purpose is right in the name: mpl to d3! A line or two and you're good to go in terms of converting your plot to d3.

Bokeh is a much larger project, at least on the surface. Even though I could generate a bar chart with the correct axes, I ran into issues that the documentation could not answer. I simply lack the time to learn Bokeh.

**Comment by aflaxman** [comment]
sorry i'm late to this party.  can you post an example of what you are trying to do and describe what specifically doesn't work for you?  The link you posted above is not working for me.  It is possible that the master branch of mpld3 does what you need:

![image](https://cloud.githubusercontent.com/assets/51236/15564643/30a873c6-22c9-11e6-89c4-c6d90871ab58.png)

[notebook showing how I did that.](http://nbviewer.jupyter.org/gist/aflaxman/53032e56290649b5fc4463241c7812bc)

**Comment by joaodlf** [comment]
Hi, 

First of all, thank for your help.

Here is what I was initially showing:
http://nbviewer.jupyter.org/gist/joaodlf/9c7f7ae95f68c5dbc088c44f5dded5e3

I'll see if I can replicate what you did later on the day, although it looks we might be working a bit differently (I'm using Dataframes and applying a few more styling options).

**Comment by etgalloway** [comment]
This is what I get when I use mpld3 from git master: http://nbviewer.jupyter.org/gist/etgalloway/5ab85467137e1db549101390db59e6d9

Issues with mpld3 master for this plot:
* The legend is not visible (it is visible if I reposition it)
* Tick labels are not rotated
* Annoying "None" in bottom left corner
* Tick label formatting is different for the vertical axis (commas in mpld3 version)

**Comment by aflaxman** [comment]
Thanks both!

Here is what that looks like for me:
![image](https://cloud.githubusercontent.com/assets/51236/15581791/b67805d0-2322-11e6-9117-a6e8b254e06b.png)

It would be cool to address all of the issues that @etgalloway identified, and likely not too hard.  Patches welcome!
